src/main/java/reactor/netty/resources/PooledConnectionProvider.java

```diff
@@ -584,7 +584,8 @@ void registerClose(PooledRef<PooledConnection> pooledRef, InstrumentedPool<Poole
 			}
 			channel.closeFuture()
 			       .addListener(ff ->
-			           pooledRef.invalidate()
+			           ((DisposableAcquire) channel.attr(OWNER).get()).pooledRef
```

Comment:
It may be necessary to do a type identification here. The owner of the channel is not always `DisposableAcquire`.

Reply:
Yes true
Fixed that yesterday
https://github.com/reactor/reactor-netty/blob/master/src/main/java/reactor/netty/resources/PooledConnectionProvider.java#L604
Thanks for the review